crates/router/src/connector/cybersource/transformers.rs

```diff
@@ -787,69 +812,217 @@ pub struct CybersourceTokenInformation {
+                            connector_mandate_id: Some(token_info.instrument_identifier.id),
+                            payment_method_id: None,
+                        });
+                let connector_response_reference_id = Some(
+                    info_response
                         .client_reference_information
-                        .map(|cref| cref.code)
-                        .unwrap_or(Some(item.response.id)),
-                    incremental_authorization_allowed: Some(
-                        status == enums::AttemptStatus::Authorized,
-                    ),
+                        .code
+                        .unwrap_or(info_response.id.clone()),
+                );
+                Ok(Self {
+                    status,
+                    response: Ok(types::PaymentsResponseData::TransactionResponse {
+                        resource_id: types::ResponseId::ConnectorTransactionId(info_response.id),
+                        redirection_data: None,
+                        mandate_reference,
+                        connector_metadata: None,
+                        network_txn_id: None,
+                        connector_response_reference_id,
+                        incremental_authorization_allowed,
+                    }),
+                    ..item.data
+                })
+            }
+            CybersourcePaymentsResponse::ErrorInformation(error_response) => Ok(Self {
+                response: Err(types::ErrorResponse {
+                    code: consts::NO_ERROR_CODE.to_string(),
+                    message: error_response
+                        .error_information
+                        .message
+                        .unwrap_or(consts::NO_ERROR_MESSAGE.to_string()),
+                    reason: error_response.error_information.reason,
+                    status_code: item.http_code,
+                    attempt_status: None,
+                    connector_transaction_id: None,
                 }),
-            },
-            ..item.data
-        })
+                ..item.data
+            }),
+        }
+    }
+}
+
+impl<F>
+    TryFrom<
+        types::ResponseRouterData<
+            F,
+            CybersourcePaymentsResponse,
+            types::PaymentsCancelData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsCancelData, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            CybersourcePaymentsResponse,
+            types::PaymentsCancelData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        match item.response {
+            CybersourcePaymentsResponse::ClientReferenceInformation(info_response) => {
+                let status = enums::AttemptStatus::foreign_from((info_response.status, false));
+                let incremental_authorization_allowed =
```

Comment:
I think we should return None here, Please confirm this with @sai-harsha-vardhan. However after cancel we will not get Authorized status

Reply:
Done